transmart-core-db/grails-app/services/org/transmartproject/db/clinical/MultidimensionalDataResourceService.groovy

```diff
@@ -468,11 +468,14 @@ class MultidimensionalDataResourceService extends AbstractDataResourceService im
         assaysByType.keySet()?.collect {it.dataTypeName}
     }
 
-    private List<MDStudy> selectStudiesWithDimensionSupport(Iterable<MDStudy> studies, DimensionImpl dimension) {
+    private List<MDStudy> selectStudiesWithDimensionSupport(Collection<MDStudy> studies, DimensionImpl dimension) {
+        if (studies.empty) {
```

Comment:
`if (studies) {` would escape possible NPE, nevertheless pass `null` is a bad idea.